File: ProductsService2/src/main/java/com/appsdeveloperblog/estore/productsservice2/data/ProductEntity.java
package com.appsdeveloperblog.estore.productsservice2.data;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;

@Entity
@Table(name="products")
@Data
public class ProductEntity implements Serializable {

    @Id
    @Column(unique=true)
    private String productId;
    @Column(unique = true)
    private String title;
    private BigDecimal price;
    private Integer quantity;
}

File: ProductsService/src/main/kotlin/com/appsdeveloperblog/estore/productsservice/data/ProductEntity.kt
package com.appsdeveloperblog.estore.productsservice.data

import jakarta.persistence.*
import java.math.BigDecimal
import java.util.UUID

@Entity
@Table(name = "products")
class ProductEntity(
    @Id
    @Column(unique = true)
    @GeneratedValue(strategy = GenerationType.AUTO)
    val productId: UUID = UUID.randomUUID(),
    @Column(nullable = false)
    val title: String = "Default Title",
    val price: BigDecimal = BigDecimal(0),
    val quantity: Int = 0
)